prow/config/secrets_agent.go

```diff
@@ -0,0 +1,76 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// SecretAgent watches a path and automatically loads the config stored
+// therein.
+type SecretAgent struct {
+	sync.Mutex
+	sc *SecretConfig
+}
+
+func (sa *SecretAgent) Start(paths ...string) error {
+	sc, err := LoadSecrets(paths...)
+	if err != nil {
+		return err
+	}
+	sa.sc = sc
+
+	go func() {
```

Comment:
Why not use `fsnotify` here with a timer backup? Did we ever move to that for the config agent @BenTheElder